### ISSUE TITLE
Adding Lights (probably using a Shader)

### ISSUE THREAD

**Issue by Sygmei** [Sygmei/ObEngine#90]

**Comment by TerensTare** [comment]
I agree that adding dependecies can be bad, but any chance [ltbl](https://github.com/hovatterz/light) can help with this? It is SFML-based so only integrating to the engine remains to do.

**Comment by Sygmei** [comment]
Yup, there is also [Candle] (https://github.com/MiguelMJ/Candle).
However, I'd like to be backend-agnostic in the future and so, to avoid picking a SFML-only depedency :)
